src/heimdall-tree/index.js

```diff
@@ -201,25 +228,23 @@ export default class HeimdallTree {
           }
 
           currentNode.stop();
+          openNodes.splice(openNodes.indexOf(currentNode), 1);
```

Comment:
This code has no effect on the critical path that the benches test. This is all about post-processing the data. While I agree this could get dicy and perhaps in the future `openNodes` could be a smarter data structure that held onto N fixed sized arrays and concated them on access, however not sure if needed at this time..